httpheader.php

```diff
@@ -68,106 +106,493 @@ public function onAfterInitialise()
+				}
+
+				// Check if the header is currently set or not
+				foreach ($oldCustomHeaders as $oldCustomHeader)
+				{
+					$found = false;
+					$customHeadersName = $oldCustomHeader->getAttribute('name');
+
+					if ($headerAndClient[0] === $customHeadersName)
+					{
+						// We found it, well done.
+						$found = true;
+						break;
+					}
+				}
+				
+				// The header wasn't found we need to create it
+				if (!$found)
+				{
+					// Generate the new header Element
+					$newHeader = $webConfigDomDoc->createElement('add');
+					$newHeader->setAttribute('name', $headerAndClient[0]);
+					$newHeader->setAttribute('value', $value);
+
+					// Append the new header
+					$customHeaders[0]->appendChild($newHeader);
+				}
+
+				$customHeadersValue = $oldCustomHeader->getAttribute('value');
+
+				if ($value === $customHeadersValue)
+				{
+					continue;
+				}
+
+				$oldCustomHeader->setAttribute('value', $value);
+			}
+		}
+
+		return $webConfigDomDoc->saveXML();
+	}
+
+	/**
+	 * Wirte the static headers.
+	 * 
+	 * @return  boolean  True on success; false on any error
 	 *
-	 * @since   1.0
+	 * @since   1.0.6
 	 */
-	private function setDefaultHeader()
+	private function writeStaticHeaders()
 	{
-		// X-Frame-Options
-		$xFrameOptions = $this->params->get('xframeoptions', 1);
+		$pathToHtaccess  = $this->getServerConfigFilePath(self::SERVER_CONFIG_FILE_HTACCESS);
+		$pathToWebConfig = $this->getServerConfigFilePath(self::SERVER_CONFIG_FILE_WEBCONFIG);
 
-		if ($xFrameOptions)
+		if (file_exists($pathToHtaccess))
 		{
-			$this->app->setHeader('X-Frame-Options', 'SAMEORIGIN');
+			$htaccessContent = $this->getHtaccessRulesForStaticHeaderConfiguration();
+
+			if (is_readable($pathToHtaccess) && !empty($htaccessContent))
+			{
+				// Write the htaccess using the Frameworks File Class
+				return File::write($pathToHtaccess, $htaccessContent);
+			}
 		}
 
-		// X-XSS-Protection
-		$xXssProtection = $this->params->get('xxssprotection', 1);
+		if (file_exists($pathToWebConfig))
+		{
+			$webConfigContent = $this->getWebConfigRulesForStaticHeaderConfiguration();
+
+			if (is_readable($pathToWebConfig) && !empty($webConfigContent))
+			{
+				// Setup and than write the web.config write using DOMDocument
+				$webConfigDomDoc = new DOMDocument();
+				$webConfigDomDoc->formatOutput = true;
+				$webConfigDomDoc->preserveWhiteSpace = false;
+				$webConfigDomDoc->loadXML($webConfigContent);
+
+				// When the return code is an integer we got the bytes and everything went well if not something broke..
+				return is_integer($webConfigDomDoc->save($pathToWebConfig)) ? true : false;
+			}
+		}
+	}
 
-		if ($xXssProtection)
+	/**
+	 * Get the configured static headers.
+	 *
+	 * @param   Registry  An Registry Object containing the plugin parameters
+	 *
+	 * @return  array  We return the array of static headers with its values.
+	 *
+	 * @since   1.0.6
+	 */
+	private function getStaticHeaderConfiguration($pluginParams = false)
```

Comment:
It's a non static function, so params are available in $this->params anyway

Reply:
depending on which state this method is used i use different sets of the params. One time (the normal case) I use $this->params for the time where we set them in the frontend.

I only get them manually in the case of onExtensionAfterSave as than this->params is not updated yet as far as i know. I might be wrong.

Reply:
Ah ok, makes sense!